packages/story-editor/src/components/publishModal/karma/publishModal.karma.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { within } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+
+describe('Publish Story Modal', () => {
+  let fixture;
+  let publishModal;
+
+  async function openPublishModal() {
+    const { publish } = fixture.editor.titleBar;
+    await fixture.events.click(publish);
+  }
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    fixture.setFlags({ enableUpdatedPublishStoryModal: true });
+    await fixture.render();
+
+    await openPublishModal();
+    publishModal = await fixture.screen.findByRole('dialog', {
+      name: /^Story details$/,
+    });
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  function getPublishModalElement(role, name) {
+    const { getByRole } = within(publishModal);
+
+    return getByRole(role, {
+      name,
+    });
+  }
```

Comment:
Only thing I would suggest is that if we use `findByRole` here, then we can `await` and get repolling. Since this is getting used for many elements, it may make it a little more robust if something ends up taking a bit longer to appear.

Reply:
thank you! 

out of pure laziness, i'm going to make this update in the next pr that was branched off of this to avoid waiting for tests to run again.